avocado/core/task/runtime.py

```diff
@@ -212,6 +213,138 @@ def get_pre_tasks_from_runnable(cls, runnable, status_server_uri=None, job_id=No
+                       sent to the destination job's status server and will
+                       make into the job's results.
+        :type job_id: str
+        :returns: RuntimeTask of the test from runnable
+        """
+        name = f'{post_runnable.kind}-{post_runnable.kwargs.get("name")}'
+        prefix = 0
+        # the human UI works with TestID objects, so we need to
+        # use it to name Task
+        task_id = TestID(prefix, name)
+        # creates the dependency task
+        task = Task(
+            post_runnable,
+            identifier=task_id,
+            status_uris=status_server_uri,
+            category="post_test",
+            job_id=job_id,
+        )
+        return cls(task, parent_task, status_server_uri, job_id)
+
+    @classmethod
+    def get_post_tasks_from_runnable(
+        cls, runnable, test_task, status_server_uri=None, job_id=None
+    ):
+        """Creates runtime tasks for postTest task from runnable
+
+        :param runnable: the "description" of what the task should run.
+        :type runnable: :class:`avocado.core.nrunner.Runnable`
+        :param test_task: PostRuntimeTask will be run after this Test task
+        :type test_task: :class:`avocado.core.task.runtime.RuntimeTask`
+        :param status_server_uri: the URIs for the status servers that this
+                                  task should send updates to.
+        :type status_server_uri: list
+        :param job_id: the ID of the job, for authenticating messages that get
+                       sent to the destination job's status server and will
+                       make into the job's results.
+        :type job_id: str
+        :returns: Pre RuntimeTasks of the dependencies from runnable
+        :rtype: list
+        """
+
+        post_test_tasks = []
+        post_plugins = TestPostDispatcher().get_extentions_by_priority()
+        for post_plugin in post_plugins:
+            runnable = Runnable(
+                post_plugin.name, None, config=test_task.task.runnable.config
+            )
+            post_task = cls.from_runnable(
+                runnable, test_task, status_server_uri, job_id
+            )
+            post_test_tasks.append(post_task)
+        return post_test_tasks
+
+    def can_run(self):
+        dependency_finished = self.are_dependencies_finished()
+        if dependency_finished:
+            for dependency in self.dependencies:
+                if dependency.result == "skip":
+                    return False
+            return True
+        return False
+
+    def get_post_plugin_tasks(self):
+        post_tasks = PostRuntimeTask.get_post_tasks_from_runnable(
+            self.task.runnable,
+            self._parent_task,
+            self._status_server_uri,
+            self._job_id,
+        )
+        for post_task in post_tasks:
+            post_task.dependencies = self.dependencies
+
+        return post_tasks
+
+
+class PostRuntimeTask(PostRuntimeTaskPrototype):
```

Comment:
Because the post-test plugin can create many tasks and the number of tasks is based on the test result as I explained in another comment. I can't create `PostRuntimeTask` during the dependency graph creation. So I created `PostRuntimeTaskPrototype` which represents all post-test tasks of one plugin inside dependency graph. And when the test ends and the result is known, then the  `PostRuntimeTaskPrototype` is replaced all `PostRuntimeTask` created by one specific post-test plugin.